src/map/ai/controllers/mob_controller.cpp

```diff
@@ -446,7 +446,7 @@ bool CMobController::TryCastSpell()
     }
 
     // Try to get an override spell from the script (if available)
-    auto possibleOverriddenSpell = luautils::OnMobMagicPrepare(PMob, PTarget, chosenSpellId);
+    auto possibleOverriddenSpell = luautils::OnMobMagicPrepare(PMob, PTarget ? PTarget : PMob, chosenSpellId);
```

Comment:
Instead of doing this inline, can you break this out into a section just before that sets PTarget to be PMob if PTarget is nullptr, leaving the lua call to OnMobMagicPrepare clean

Reply:
I was wary to change ptarget in the controller. I think it needs to be null when out of combat

Reply:
Then don't change it in the controller, add:

```
auto* PSpellTarget = PTarget ? PTarget : PMob;
```

and fill in the lua binding appropriately